kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/TemplateTest.java

```diff
@@ -212,4 +213,16 @@ public void testProcessParameterizedNumberTemplate() throws IOException {
     KubernetesList list = client.templates().withName("tmpl1").processLocally(map);
     assertListIsServiceWithPort8080(list);
   }
+
+  @Test
+  public void testShouldNotThrowNullPointerException() {
+    server.expect().withPath("/oapi/v1/namespaces/test/templates/tmpl1").andReturn(200, new TemplateBuilder()
+      .withNewMetadata().withName("tmpl1").endMetadata()
+      .withParameters(new ParameterBuilder().withName("key").build())
+      .build()).once();
+    OpenShiftClient client = server.getOpenshiftClient();
+    KubernetesList list = client.templates().withName("tmpl1").processLocally(Collections.singletonMap("key", (String) null));
+    assertNotNull(list);
```

Comment:
Can this test be a bit more descriptive? 

- when it shouldn't throw NPE? maybe method name needs a bit more love :)
- if you provided default value to `""` it would be good to cover that as it becomes a contract now